worker/sshserver/manifold.go

```diff
@@ -75,14 +76,19 @@ func (config ManifoldConfig) startWrapperWorker(context dependency.Context) (wor
 		_ = stTracker.Done()
 		return nil, errors.Trace(err)
 	}
-
-	sysState, err := statePool.SystemState()
+	systemState, err := statePool.SystemState()
 	if err != nil {
+		_ = stTracker.Done()
+		return nil, errors.Trace(err)
+	}
+	sshAuthenticator, err := sshauth.NewAuthenticator(statePool, config.Logger)
```

Comment:
This seems like something we should construct in worker.go because the Authenticator does not come from the manifold engine.

Reply:
Valid reasons, I'm okay with either way but I'll wait for others to see if there is a clear line.